node/src/reactor/participating.rs

```diff
@@ -608,129 +603,10 @@ impl reactor::Reactor for Reactor {
-                effects.extend(reactor::wrap_effects(Into::into, new_effects));
-                new_header
-            } else {
-                return Err(Error::UnexpectedLatestBlockHeader {
-                    latest_block_header: Box::new(latest_block_header),
-                });
-            }
-        } else {
-            match chainspec.protocol_config.activation_point {
-                ActivationPoint::Genesis(genesis_timestamp) => {
-                    // Do not run genesis on a node which is not protocol version 1.0.0
-                    if chainspec.protocol_config.version != ProtocolVersion::V1_0_0 {
-                        return Err(Error::GenesisNeedsProtocolVersion1_0_0 {
-                            chainspec_protocol_version: chainspec.protocol_config.version,
-                        });
-                    }
-
-                    // Check that no blocks exist in storage so we don't try to run genesis again on
-                    // an existing blockchain.
-                    if let Some(highest_block_header) = storage.read_highest_block_header()? {
-                        return Err(Error::CannotRunGenesisOnPreExistingBlockchain {
-                            highest_block_header: Box::new(highest_block_header),
-                        });
-                    }
-                    let GenesisSuccess {
-                        post_state_hash,
-                        execution_effect,
-                    } = contract_runtime.commit_genesis(chainspec)?;
-                    info!("genesis chainspec name {}", chainspec.network_config.name);
-                    info!("genesis state root hash {}", post_state_hash);
-                    trace!(%post_state_hash, ?execution_effect);
-                    let initial_pre_state = ExecutionPreState::new(
-                        0,
-                        post_state_hash,
-                        Default::default(),
-                        Default::default(),
-                    );
-                    let (new_header, new_effects) = create_immediate_switch_block(
-                        &mut contract_runtime,
-                        &mut storage,
-                        effect_builder,
-                        chainspec,
-                        initial_pre_state,
-                        genesis_timestamp,
-                        EraId::from(0u64),
-                    )?;
-                    effects.extend(reactor::wrap_effects(Into::into, new_effects));
-                    new_header
-                }
-                ActivationPoint::EraId(upgrade_era_id) => {
-                    return Err(Error::NoSuchSwitchBlockHeaderForUpgradeEra { upgrade_era_id });
-                }
-            }
-        };
-
         let (block_proposer, block_proposer_effects) = BlockProposer::new(
             registry.clone(),
             effect_builder,
-            latest_block_header.height() + 1,
+            latest_switch_block_header.height() + 1,
```

Comment:
I think that's also not necessarily a switch block? If we did do fast sync, it's just the latest block?

Reply:
Renamed

Reply:
Did you? I still see `latest_switch_block_header`.